File: PlaceFeedsApp/ClientApp/app/services/weather/weather.service.ts
import { Injectable } from '@angular/core';
import { Http, Response } from '@angular/http';
import { Observable } from 'rxjs/observable';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/do';
import 'rxjs/add/operator/catch';

@Injectable()
export class WeatherService {
    //Uses OpenWeatherMap

    constructor(private _http: Http) { }

    getWeatherData(lat: number, lon: number): Observable<any> {

        return this._http.get(`Weather/GetWeatherData/?latitude=${lat}&longitude=${lon}`)
            .map((response: Response) => response.json());
    }

};

File: PlaceFeedsServices/WeatherService/WeatherService.cs
using PlaceFeedsServices.ApiKeyService;
using PlaceFeedsServices.Enums;
using System;
using System.Net.Http;
using System.Threading.Tasks;

namespace PlaceFeedsServices.WeatherService
{
    public class WeatherService : IWeatherService
    {
        private readonly IApiKeyService _apiKeyService;

        public WeatherService(IApiKeyService apiKeyService)
        {
            _apiKeyService = apiKeyService;
        }

        public async Task<string> GetWeatherData(float latitude, float longitude)
        {
            using (HttpClient client = new HttpClient())
            {
                string apiKey = _apiKeyService.GetApiKey(ApiType.Weather);

                client.BaseAddress = new Uri("http://api.openweathermap.org/data/2.5/");
                var response = await client.GetAsync($"forecast?lat={latitude}&lon={longitude}&APPID={apiKey}");
                response.EnsureSuccessStatusCode();

                return await response.Content.ReadAsStringAsync();
            }
        }

    }
}
